src/main/resources/avro/variants.avdl

```diff
@@ -120,30 +168,66 @@ record Call {
   union { null, string } callSetName = null;
 
   /**
-  The genotype of this variant call. Each value represents either the value
-  of the referenceBases field or is a 1-based index into alternateBases.
+  The genotype of this variant call. Must be null if there is an associated
+  `Allele`, which implies a uniform genotype of that `Allele`.
+
+  A 0 value represents the reference allele of the associated `Variant`. Any
+  other value is a 1-based index into the alternate alleles of the associated
+  `Variant`.
+
   If a variant had a referenceBases field of "T", an alternateBases
   value of ["A", "C"], and the genotype was [2, 1], that would mean the call
   represented the heterozygous value "CA" for this variant. If the genotype
   was instead [0, 1] the represented value would be "TA". Ordering of the
   genotype values is important if the phaseset field is present.
+
+  If the API server supports the "classic" mode, and this `Call` is associated
+  with a `Variant`, this field must not be null.
   */
-  array<int> genotype = [];
+  union { null, array<int> } genotype = null;
 
   /**
-  If this field is present, this variant call's genotype ordering implies
+  If specified, gives the total number of copies of whatever is being called. If
+  the `Call` is associated only with a `Variant`, it is the total copy number of
+  all `Allele`s of a `Variant`. If the `Call` is associated with only an
+  `Allele`, it is the total copy number of that `Allele` (including copies
+  associated with `Variant`s). If the `Call` is associated with an `Allele` and
+  a `Variant`, it is the copy number of that `Allele` in that `Variant`.
+  */
+  union { null, double } totalCopies = null;
+
+  /**
+  If this field is a string, this variant call's genotype ordering implies
   the phase of the bases and is consistent with any other variant calls on
-  the same contig which have the same phaseset value.
+  the same contig which have the same phaseset string.
+
+  If this field is an array, each entry is associated with the corresponding
+  entry in the `genotype` array, or with an instance of the associated `Allele`
+  if one exists. Allele instances that are associated with the same `phaseset`
+  string occur on the same molecule of DNA.
+
+  For example, if one `Call` has `genotype` [0, 1], another is associated with
+  `Allele` 3, and both have a `phaseset` of ["maternal", "paternal"], then one
+  DNA molecule carries 0, 3 and another carries 1, 3.
+
+  If the API server supports the "classic" mode, and this `Call` is associated
+  with a `Variant`, this field must not be an array. If the API server supports
+  the "graph" mode and not the "classic" mode, it must not return strings here.
   */
-  union { null, string } phaseset = null;
+  union { null, string, array<string> } phaseset = null;
```

Comment:
On Wed, Feb 11, 2015 at 10:07 AM, Mark Diekhans <notifications@github.com>
wrote:

> In src/main/resources/avro/variants.avdl
> <https://github.com/ga4gh/schemas/pull/238#discussion_r24517372>:
>
> >    */
> > -  union { null, string } phaseset = null;
> > +  union { null, string, array<string> } phaseset = null;
>
> +1 on this Frank Austin Nothaft <notifications@github.com> writes:
>

No, because this breaks backwards compatibility. We're all for breaking
backwards compatibility down the road - but it needs to be in separate pull
requests. This is supposed to be a minimal set of changes.


> … <#14b79d2bba4db338_>
>  In src/main/resources/avro/variants.avdl: > */ > - union { null, string }
> phaseset = null; > + union { null, string, array<string> } phaseset = null;
> Why not just make this an array<string>? I really feel that encoding the
> behavior of the variant (e.g., allelic vs. genotypic) via type polymorphism
> is just going to make things difficult to use for developers. — Reply to
> this email directly or view it on GitHub.*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ga4gh/schemas/pull/238/files#r24517372>.
>